### PR TITLE
Pitch shifted collision sounds

### PR DESCRIPTION
**Pull request by howard-stearns**
Shift the pitch of collision sounds based on the size of the entity that the sound is attached to.
(Big stuff is deeper.)
The largest dimension of each entity bounding box is some multiple times a standard size (currently 0.2m).  We drop the octave of the pitch proportional to the (ln of) that multiple.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 622a1e30688982863eb73ef1f5edc229c2f5db6f available at: https://gist.github.com/120cf948940cbd9914ed<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5019.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5019.dmg

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -1158,7 +1159,30 @@ void EntityTreeRenderer::playEntityCollisionSound(const QUuid& myNodeID, EntityT
...
-    AudioInjector* injector = new AudioInjector(sound.data(), options);
+
+    // Shift the pitch down by ln(1 + (size / COLLISION_SIZE_FOR_STANDARD_PITCH)) / ln(2)
+    const float COLLISION_SIZE_FOR_STANDARD_PITCH = 0.2f;
+    QByteArray samples = sound->getByteArray();
+    soxr_io_spec_t spec = soxr_io_spec(SOXR_INT16_I, SOXR_INT16_I);
+    soxr_quality_spec_t qualitySpec = soxr_quality_spec(SOXR_MQ, 0);
+    const int channelCount = sound->isStereo() ? 2 : 1;
+    const float factor = log(1.0f + (entity->getMaximumAACube().getLargestDimension() / COLLISION_SIZE_FOR_STANDARD_PITCH)) / log(2);
+    const int HIFI_SAMPLE_RATE = 24000; //AudioConstants::SAMPLE_RATE
```
Comment (PhilipRosedale) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: Is there a way to get the actual sample rate from the constants, as you suggest in the comment?

**Comment by PhilipRosedale** [comment]
that sounds really great!  At least for resizing dice.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0837c016c52237ca6dcffdac98b8ca4d514bf4af available at: https://gist.github.com/120cf948940cbd9914ed<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5019.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5019.dmg

**Pull request merged**